### DISCUSSION TITLE
useSource is marked as depreciated

### DISCUSSION THREAD

**Discussion by joshuaellis** [sanity-io/sanity#3335] [Sanity Studio Support] [answered]
I’m trying to use useSource in my V3 plugin as described here – https://beta.sanity.io/docs/platform/studio/v2-to-v3/plugins#migrating-part-imports but TS says this method is depreciated, is this a mistake?

See image:

<img width="485" alt="image" src="https://user-images.githubusercontent.com/37798644/173610998-de7daf3f-2bd6-41ca-84ea-ede1a025b989.png">

I would have given you a permalink, but no idea where it is in your codebase (sorry)

**Comment by p10e** [comment, marked as answer]
Thanks for reporting, @joshuaellis! Looks like we'll have to update our documentation. Could you try using these hooks instead of `useSource()`?

```
// new hooks
import {
  useClient,
  useDataset,
  useProjectId,
  useSchema,
  useTemplates,
  useTools,
} from 'sanity'

const client = useClient({apiVersion: '2022-05-01'})
const dataset = useDataset()
const projectId = useProjectId()
const schema = useSchema()
const templates = useTemplates()
const tools = useTools()
```

**Reply by joshuaellis** [reply]
Yep, `useClient` was what I needed for my goal. Thanks!
